src/command.c

```diff
@@ -179,8 +190,10 @@ handle_command_stop (const struct subcommand *sub,
 	cc_oci_node_dump(root);
 #endif /*DEBUG*/
 
-	g_node_children_foreach(root, G_TRAVERSE_ALL,
-	    (GNodeForeachFunc)process_config_stop, (gpointer)config);
+	if (! cc_oci_process_config(root, config, stop_spec_handlers)) {
```

Comment:
Instead of open coding a tree traversing routine, the typical way of tracking errors with a GLib `*_foreach()` routine is to create a:
```
struct config_cookie {
          gboolean failed
          struct cc_oci_config *config
}
}
```

And passing a `config_cookie` pointer to `g_node_children_foreach()`. Your `GNodeForEachFunc` callback is a wrapper around `process_config_stop()` that only sets `config_cookie->failed` to true when `process_config_stop()` fails.